relay-dynamic-config/src/metrics.rs

```diff
@@ -545,4 +657,99 @@ mod tests {
         let mapping: TagMapping = serde_json::from_str(json).unwrap();
         assert!(mapping.metrics[0].compiled().is_match("d:spans/foo"));
     }
+
+    fn templates() -> MetricExtractionTemplates {
+        serde_json::from_value::<MetricExtractionTemplates>(serde_json::json!({
+            "templates": {
+                "template1": {
+                    "isEnabled": false,
+                    "metrics": [{
+                        "category": "transaction",
+                        "mri": "c:metric1/counter@none",
+                    }],
+                    "tags": [
+                        {
+                            "metrics": ["c:metric1/counter@none"],
+                            "tags": [{
+                                "key": "tag1",
+                                "value": "value1"
+                            }]
+                        }
+                    ]
+                },
+                "template2": {
+                    "isEnabled": true,
+                    "metrics": [{
+                        "category": "transaction",
+                        "mri": "c:metric2/counter@none",
+                    }],
+                    "tags": [
+                        {
+                            "metrics": ["c:metric2/counter@none"],
+                            "tags": [{
+                                "key": "tag2",
+                                "value": "value2"
+                            }]
+                        }
+                    ]
+                }
+            }
+        }))
+        .unwrap()
+    }
+
+    #[test]
+    fn metrics_extraction_global_defaults() {
```

Comment:
Can we add tests like these for merging tags?

Reply:
IMO it would not add much value, the group can be either enabled or disabled, so you either get all metrics and all tags, or none.